nautobot/dcim/models/device_components.py

```diff
@@ -499,11 +499,12 @@ class BaseInterface(RelationshipModel):
     class Meta:
         abstract = True
 
-    def save(self, *args, **kwargs):
-
+    def clean(self):
         # Remove untagged VLAN assignment for non-802.1Q interfaces
-        if not self.mode:
-            self.untagged_vlan = None
+        if not self.mode and self.untagged_vlan is not None:
+            raise ValidationError({"untagged_vlan": "Mode must be set when specifiying unagged_vlan"})
```

Comment:
```suggestion
            raise ValidationError({"untagged_vlan": "Mode must be set when specifying untagged_vlan"})
```

Reply:
Sheesh and to think I was once a Spelling Bee champion.